docs/how-to/vector-stores/hybrid-search.md

```diff
@@ -0,0 +1,169 @@
+- **Sparse retrieval**: Using sparse embeddings for lexical/keyword matching
+
+By combining these approaches, hybrid search can overcome the limitations of each individual method.
+
+## Implementing Hybrid Search in Ragbits
+
+Ragbits provides several strategies for combining results from different vector stores:
+
+### 1. Set Up Multiple Vector Stores
+
+First, create vector stores with different embedding types:
+
+```python
+from ragbits.core.embeddings.dense import DenseEmbedder
+from ragbits.core.embeddings.sparse import SparseEmbedder
+from ragbits.core.vector_stores.in_memory import InMemoryVectorStore
+from ragbits.core.vector_stores.base import EmbeddingType
+
+# Dense embedder for semantic search
+dense_embedder = DenseEmbedder(model_name="sentence-transformers/all-MiniLM-L6-v2")
+dense_store = InMemoryVectorStore(
+    embedder=dense_embedder,
+    embedding_type=EmbeddingType.TEXT
+)
+
+# Sparse embedder for lexical search
+sparse_embedder = SparseEmbedder(vocabulary_size=10000)
+sparse_store = InMemoryVectorStore(
+    embedder=sparse_embedder,
+    embedding_type=EmbeddingType.TEXT
+)
+```
+
+### 2. Choose a Hybrid Retrieval Strategy
+
+Ragbits provides several strategies for combining results:
+
+```python
+from ragbits.core.vector_stores.hybrid_strategies import (
+    OrderedHybridRetrivalStrategy,
+    ReciprocalRankFusion,
+    DistributionBasedScoreFusion
+)
+
+# Choose one of the available strategies
+strategy = ReciprocalRankFusion()
+```
+
+Available strategies:
+- **OrderedHybridRetrivalStrategy**: Orders results by score and deduplicates them
+- **ReciprocalRankFusion**: Combines results based on their ranks rather than raw scores
+- **DistributionBasedScoreFusion**: Normalizes scores based on their distribution before combining
+
+### 3. Perform Hybrid Search
+
+```python
+async def hybrid_search(query: str, k: int = 5):
+    # Get results from both stores
+    dense_results = await dense_store.retrieve(query)
+    sparse_results = await sparse_store.retrieve(query)
+
+    # Combine results using the chosen strategy
+    combined_results = strategy.join([dense_results, sparse_results])
+
+    # Return top k results
+    return combined_results[:k]
+```
+
+## Advanced Hybrid Search Techniques
+
+### Weighted Combination
+
+You can implement custom weighting between dense and sparse results:
+
+```python
+from ragbits.core.vector_stores.hybrid_strategies import HybridRetrivalStrategy
+from ragbits.core.vector_stores.base import VectorStoreResult
+
+class WeightedHybridStrategy(HybridRetrivalStrategy):
+    def __init__(self, dense_weight: float = 0.7, sparse_weight: float = 0.3):
+        self.dense_weight = dense_weight
+        self.sparse_weight = sparse_weight
+
+    def join(self, results: list[list[VectorStoreResult]]) -> list[VectorStoreResult]:
+        if len(results) != 2:
+            raise ValueError("WeightedHybridStrategy expects exactly 2 result lists")
+
+        dense_results, sparse_results = results
```

Comment:
There is no reason to assume that the first results are dense and the second are sparse. You would need to look at the actual vector type to distinguish between them. Or even better - instead of having separate weights for "sparse" and "dense" just take a list of weights, when the i-th weight would represent the weight for the i-th result list.